From dfc4e1801dff7037ba0700ec142c331f91f811c1 Mon Sep 17 00:00:00 2001
From: Naveen Gangam <ngangam@apache.org>
Date: Thu, 18 May 2017 10:08:02 -0400
Subject: [PATCH 1123/1164] CDH-51849: HIVE-16697 Schema table validator
 should return a sorted list of missing tables
 (Naveen Gangam, reviewed by Aihua Xu)

Conflicts:
	beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java

Change-Id: I852ce651146e28b65af8caea5c9c27ec926209a5
---
 .../org/apache/hive/beeline/HiveSchemaTool.java    |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java b/beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java
index 83b5fc6..bb6e7e3 100644
--- a/beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java
+++ b/beeline/src/java/org/apache/hive/beeline/HiveSchemaTool.java
@@ -62,6 +62,7 @@
 import java.sql.DatabaseMetaData;
 import java.sql.PreparedStatement;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
@@ -783,6 +784,7 @@ boolean validateSchemaTables(Connection conn) throws HiveMetaException {
     // now diff the lists
     schemaTables.removeAll(dbTables);
     if (schemaTables.size() > 0) {
+      Collections.sort(schemaTables);
       System.err.println("Table(s) [ " + Arrays.toString(schemaTables.toArray())
           + " ] are missing from the metastore database schema.");
       System.out.println("Schema table validation failed!!!");
-- 
1.7.9.5

